monsteer/plugin/spikeReport.cpp

```diff
@@ -248,19 +248,20 @@ void SpikeReport::clear( const float startTime, const float endTime )
                    _spikes.upper_bound( endTime ));
 }
 
-void SpikeReport::_onSpikes( const zeroeq::FBEvent& event )
+void SpikeReport::_onSpikes( ConstSpikesEventPtr event )
 {
-    LBASSERT( event.getTypeIdentifier() == EVENT_SPIKES );
-    const SpikeMap& spikes = deserializeSpikes( event );
-    _incoming.insert( spikes.begin(), spikes.end( ));
+    for( size_t i = 0; i < event->getSpikes().size( ); ++i )
+    {
+        const Spike& spike = event->getSpikes()[i];
+        _incoming.insert( std::make_pair( spike.getTime(), spike.getCell( )));
```

Comment:
Spikes should be sorted at origin, use the insert with hint to insert at the end for performance.